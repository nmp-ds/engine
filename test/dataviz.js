import { expect, test } from 'vitest';

import { setup } from './_helpers.js';

setup();

test('it generates css based on semantic dataviz tokens', async ({ uno }) => {
  const classes = [
    'dv-s-fill-border-warning',
    'dv-s-stroke-border-warning',
    'dv-s-border-warning',
    'dv-s-border-l-warning',
    'dv-s-border-r-warning',
    'dv-s-border-b-warning',
    'dv-s-border-t-warning',
    'dv-s-border-y-warning',
    'dv-s-border-x-warning',
    'dv-s-stroke-line-warning',
    'dv-s-line-warning',
    'dv-s-outline-neutral',
    'dv-s-outline-negative',
    'dv-s-outline-warning',
    'dv-s-outline-positive',
    'dv-s-outline-category6',
    'dv-s-fill-bg-neutral',
    'dv-s-bg-primary',
    'dv-s-bg-neutral',
    'dv-s-bg-negative',
    'dv-s-bg-warning',
    'dv-s-bg-positive',
    'dv-s-bg-category6',
    'dv-s-stroke-icon-category7',
    'dv-s-fill-icon-category7',
    'dv-s-icon-category7',
    'dv-s-icon-category8',
    'dv-s-icon-negative',
    'dv-s-icon-neutral',
    'dv-s-icon-positive',
    'dv-s-icon-warning',
    'dv-s-fill-text-category7',
    'dv-s-text-category7',
    'dv-s-text-category8',
    'dv-s-text-negative',
    'dv-s-text-neutral',
    'dv-s-text-positive',
    'dv-s-text-warning',
    'dv-s-chartline',
    'dv-s-fill-chartline',
    'dv-s-stroke-chartline',
    'dv-s-chartline-subtle',
    'dv-s-fill-chartline-subtle',
    'dv-s-stroke-chartline-subtle',
    'dv-s-chartbg',
    'dv-s-fill-chartbg',
    'dv-s-stroke-chartbg',
    'dv-s-charttext',
    'dv-s-fill-charttext',
    'dv-s-stroke-charttext',
    'dv-s-charttext-subtle',
    'dv-s-fill-charttext-subtle',
    'dv-s-stroke-charttext-subtle',
  ];
  const antiClasses = [
    'dv-s-background',
    'dv-s-background-positive-selected',
    'dv-s-font',
    'dv-s-font-link-active',
    'dv-s-bordercolor',
    'dv-s-bordercolor-l-disabled',
  ];
  const { css } = await uno.generate([...classes, ...antiClasses]);
  expect(css).toMatchSnapshot();
});

test('it generates css based on semantic dataviz tokens with alpha channel', async ({ uno }) => {
  const classes = [
    'dv-s-border-warning/90',
    'dv-s-fill-border-negative/90',
    'dv-s-stroke-line-negative/90',
    'dv-s-bg-positive/100',
    'dv-s-fill-bg-neutral/15',
    'dv-s-outline-category7/5',
    'dv-s-icon-category8/99',
    'dv-s-fill-icon-negative/1',
    'dv-s-stroke-icon-neutral/0',
    'dv-s-text-positive/4',
    'dv-s-fill-text-warning/25',
  ];
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('it should not generate css with incorrect alpha channel values', async ({ uno }) => {
  const antiClasses = [
    'dv-s-bg/900',
    'dv-s-bg/',
    'dv-s-bg/101',
    'dv-s-bg/001',
    'dv-s-bg/1000',
    'dv-s-bg/00',
    'dv-s-bg/01',
    'dv-s-bg-positive-selected-highlight/00',
    'dv-s-text/001',
    'dv-s-text-link-active/5566',
    'dv-s-border/1000',
    'dv-s-border-l/505',
  ];
  const { css } = await uno.generate(antiClasses);
  expect(css).toHaveLength(0);
});

test('pseudo', async ({ uno }) => {
  const classes = [
    'focus:dv-s-icon-category7-highlight',
    'focus:dv-s-fill-icon-category8-highlight',
    'focus:dv-s-stroke-icon-negative-highlight',
    'focus:dv-s-icon-neutral-highlight',
    'focus:dv-s-icon-positive-highlight',
    'focus:dv-s-icon-warning-highlight',
    'hover:dv-s-border-primary-highlight',
    'hover:dv-s-bg-secondary-highlight',
    'hover:dv-s-icon-negative-highlight',
    'hover:dv-s-icon-neutral-highlight',
    'hover:dv-s-stroke-line-neutral-highlight',
    'hover:dv-s-line-neutral-highlight',
    'hover:dv-s-text-positive-highlight',
    'hover:dv-s-text-warning-highlight',
  ];
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});
