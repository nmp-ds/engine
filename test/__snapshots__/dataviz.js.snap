// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it generates css based on semantic dataviz tokens 1`] = `
"/* layer: default */
.dv-s-svgfill-neutral{fill:var(--w-dv-s-color-background-neutral);}
.dv-s-bg-category6{background-color:var(--w-dv-s-color-background-category6);}
.dv-s-bg-negative{background-color:var(--w-dv-s-color-background-negative);}
.dv-s-bg-neutral{background-color:var(--w-dv-s-color-background-neutral);}
.dv-s-bg-positive{background-color:var(--w-dv-s-color-background-positive);}
.dv-s-bg-warning{background-color:var(--w-dv-s-color-background-warning);}
.dv-s-svgline-warning{stroke:var(--w-dv-s-color-line-warning);}
.dv-s-svgborder-warning{fill:var(--w-dv-s-color-border-warning);}
.dv-s-border-b-warning{border-bottom-color:var(--w-dv-s-color-border-warning);}
.dv-s-border-l-warning{border-left-color:var(--w-dv-s-color-border-warning);}
.dv-s-border-r-warning{border-right-color:var(--w-dv-s-color-border-warning);}
.dv-s-border-t-warning{border-top-color:var(--w-dv-s-color-border-warning);}
.dv-s-border-warning{border-color:var(--w-dv-s-color-border-warning);}
.dv-s-border-x-warning{border-left-color:var(--w-dv-s-color-border-warning);border-right-color:var(--w-dv-s-color-border-warning);}
.dv-s-border-y-warning{border-top-color:var(--w-dv-s-color-border-warning);border-bottom-color:var(--w-dv-s-color-border-warning);}
.dv-s-outline-category6{outline-color:var(--w-dv-s-color-border-category6);}
.dv-s-outline-negative{outline-color:var(--w-dv-s-color-border-negative);}
.dv-s-outline-neutral{outline-color:var(--w-dv-s-color-border-neutral);}
.dv-s-outline-positive{outline-color:var(--w-dv-s-color-border-positive);}
.dv-s-outline-warning{outline-color:var(--w-dv-s-color-border-warning);}
.dv-s-svgtext-category7{fill:var(--w-dv-s-color-text-category7);}
.dv-s-text-category7{color:var(--w-dv-s-color-text-category7);}
.dv-s-text-category8{color:var(--w-dv-s-color-text-category8);}
.dv-s-text-negative{color:var(--w-dv-s-color-text-negative);}
.dv-s-text-neutral{color:var(--w-dv-s-color-text-neutral);}
.dv-s-text-positive{color:var(--w-dv-s-color-text-positive);}
.dv-s-text-warning{color:var(--w-dv-s-color-text-warning);}
.dv-s-svgiconfill-category7{fill:var(--w-dv-s-color-icon-category7);}
.dv-s-svgiconstroke-category7{stroke:var(--w-dv-s-color-icon-category7);}
.dv-s-icon-category7{color:var(--w-dv-s-color-icon-category7);}
.dv-s-icon-category8{color:var(--w-dv-s-color-icon-category8);}
.dv-s-icon-negative{color:var(--w-dv-s-color-icon-negative);}
.dv-s-icon-neutral{color:var(--w-dv-s-color-icon-neutral);}
.dv-s-icon-positive{color:var(--w-dv-s-color-icon-positive);}
.dv-s-icon-warning{color:var(--w-dv-s-color-icon-warning);}"
`;

exports[`it generates css based on semantic dataviz tokens with alpha channel 1`] = `
"/* layer: default */
.dv-s-svgfill-neutral\\/15{fill:rgba(var(--w-s-rgb-background-neutral), 0.15);}
.dv-s-bg-positive\\/100{background-color:rgba(var(--w-s-rgb-background-positive), 1);}
.dv-s-svgline-negative\\/90{stroke:rgba(var(--w-s-rgb-line-negative), 0.9);}
.dv-s-svgborder-negative\\/90{fill:rgba(var(--w-s-rgb-border-negative), 0.9);}
.dv-s-border-warning\\/90{border-color:rgba(var(--w-s-rgb-border-warning), 0.9);}
.dv-s-outline-category7\\/5{outline-color:rgba(var(--w-s-rgb-border-category7), 0.05);}
.dv-s-svgtext-warning\\/25{fill:rgba(var(--w-s-rgb-text-warning), 0.25);}
.dv-s-text-positive\\/4{color:rgba(var(--w-s-rgb-text-positive), 0.04);}
.dv-s-svgiconfill-negative\\/1{fill:rgba(var(--w-s-rgb-icon-negative), 0.01);}
.dv-s-svgiconstroke-neutral\\/0{stroke:rgba(var(--w-s-rgb-icon-neutral), 0);}
.dv-s-icon-category8\\/99{color:rgba(var(--w-s-rgb-icon-category8), 0.99);}"
`;

exports[`pseudo 1`] = `
"/* layer: default */
.hover\\:dv-s-bg-secondary-hover:hover{background-color:var(--w-dv-s-color-background-secondary-hover);}
.hover\\:dv-s-svgline-neutral-hover:hover{stroke:var(--w-dv-s-color-line-neutral-hover);}
.hover\\:dv-s-border-primary-hover:hover{border-color:var(--w-dv-s-color-border-primary-hover);}
.hover\\:dv-s-text-positive-hover:hover{color:var(--w-dv-s-color-text-positive-hover);}
.hover\\:dv-s-text-warning-hover:hover{color:var(--w-dv-s-color-text-warning-hover);}
.focus\\:dv-s-svgiconfill-category8-focus:focus{fill:var(--w-dv-s-color-icon-category8-focus);}
.focus\\:dv-s-svgiconstroke-negative-focus:focus{stroke:var(--w-dv-s-color-icon-negative-focus);}
.hover\\:dv-s-icon-negative-hover:hover{color:var(--w-dv-s-color-icon-negative-hover);}
.hover\\:dv-s-icon-neutral-hover:hover{color:var(--w-dv-s-color-icon-neutral-hover);}
.focus\\:dv-s-icon-category7-focus:focus{color:var(--w-dv-s-color-icon-category7-focus);}
.focus\\:dv-s-icon-neutral-focus:focus{color:var(--w-dv-s-color-icon-neutral-focus);}
.focus\\:dv-s-icon-positive-focus:focus{color:var(--w-dv-s-color-icon-positive-focus);}
.focus\\:dv-s-icon-warning-focus:focus{color:var(--w-dv-s-color-icon-warning-focus);}"
`;
